src/checker.ml

```diff
@@ -958,11 +958,25 @@ let view_is_burrow_liquidatable (burrow_id, state: burrow_id * checker) : bool =
   let burrow = burrow_touch state.parameters burrow in
   burrow_is_liquidatable state.parameters burrow
 
-let view_current_liquidation_auction_minimum_bid ((), state: unit * checker) : view_current_liquidation_auction_minimum_bid_result =
+let view_current_liquidation_auction_details ((), state: unit * checker) : view_current_liquidation_auction_details_result =
   assert_checker_invariants state;
   let auction = liquidation_auction_get_current_auction state.liquidation_auctions in
-  { auction_id = auction.contents
-  ; minimum_bid = liquidation_auction_current_auction_minimum_bid auction
+  let current_bid, blocks, seconds = match auction.state with
+    | Descending _ -> (None: (bid option)), (None: (Ligo.int option)), (None: (Ligo.int option))
+    | Ascending (current_bid, bid_time_seconds, bid_level) ->
+      (
+        Some current_bid,
+        Some (Ligo.sub_nat_nat (Ligo.add_nat_nat bid_level max_bid_interval_in_blocks) !Ligo.Tezos.level),
+        Some (Ligo.sub_timestamp_timestamp (Ligo.add_timestamp_int bid_time_seconds max_bid_interval_in_seconds) !Ligo.Tezos.now)
```

Comment:
I now wonder what the values of `Tezos.level` and `Tezos.now` are in an offline environment where the view is executed :thinking: Might be a better idea to instead give "safer" values such as
```ocaml
last_block_to_bid_in     = Ligo.add_nat_nat bid_level max_bid_interval_in_blocks
last_timestamp_to_bid_in = Ligo.add_timestamp_int bid_time_seconds max_bid_interval_in_seconds
```
To me it's fine either way, but we should probably test at some point that these two work as expected in a non-OCaml environment. In the long run we'd probably prefer the stable versions.

Reply:
That's a good point. After looking into it, it seems like it depends on how you execute the view. pytezos, for example, uses it's own Michelson interpreter along the contract state at whatever level you specify. If you use the interpreter on a tezos node, I think that `Tezos.level` and `Tezos.now` will match whatever level the node is synced to.

In practice you would need the node to be fully synced to make a bid, so I don't think the current approach is too inaccurate. Your suggestion would definitely leave less room for interpretation since it forces the caller to think about what `Tezos.now` actually is.

Reply:
Right, I thought so. Then I'd certainly suggest we switch to the clearer versions, either in this PR or in the next.

Reply:
Okay great, I think I'll go ahead and merge this one then look into updating the logic in a follow-up PR. This seems to be a bit of a question of convenience (avoiding extra user logic and calls to the node to check the level / time) vs. ensuring accuracy, and I'm honestly not sure which approach would be most valuable to a prospective liquidation auction user :thinking:

Reply:
Yeah, I guess it is a little extra logic for the call sites, but I think the extra clarity and unambiguity is certainly worth it. If users are participating in liquidation auctions I expect them to be able to figure out the current level/timestamp themselves.